### PR TITLE
Process lfp from broadband

### PR DESCRIPTION
**Pull request by leoscholl**
improves speed and accuracy of lfp preprocessing

### CODE REVIEW

File: aopy/data/bmi3d.py [hunk 1]
```diff
@@ -197,6 +197,149 @@ def proc_ecube_data(data_path, data_source, result_filepath, result_name='broadb
...
+    # Create an hdf dataset
+    hdf = h5py.File(result_filepath, 'a') # should append existing or write new?
+    dset = hdf.create_dataset('lfp_data', (lfp_samples, n_channels), dtype=dtype)
+
+    # Filter broadband data into LFP directly into the hdf file
+    n_lfp_samples = 0
+    for broadband_chunk in load_ecube_data_chunked(ecube_filepath, 'Headstages', chunksize=chunksize):
+        lfp_chunk = precondition.filter_lfp(broadband_chunk, samplerate, **filter_kwargs)
+        chunk_len = lfp_chunk.shape[0]
+        print(n_lfp_samples, n_lfp_samples+chunk_len, lfp_samples)
```
Comment (rcanfield2) [inline aopy/data/bmi3d.py:324]: necessary?

File: aopy/data/bmi3d.py [hunk 2]
```diff
@@ -197,6 +197,149 @@ def proc_ecube_data(data_path, data_source, result_filepath, result_name='broadb
...
+        dset -= np.mean(dset, axis=0, dtype=dtype) # hopefully this isn't constrained by memory
+    hdf.close()
+
+    # Append the lfp metadata to the file
+    lfp_metadata = metadata
+    lfp_metadata['lfp_samplerate'] = lfp_samplerate # for backwards compatibility
+    lfp_metadata['samplerate'] = lfp_samplerate
+    lfp_metadata['n_samples'] = lfp_samples
+    lfp_metadata['low_cut'] = 500
+    lfp_metadata['buttord'] = 4
```
Comment (rcanfield2) [inline aopy/data/bmi3d.py:338]: Do these need to be updated based on **filter_kwargs?

Comment (rcanfield2) [inline aopy/data/bmi3d.py:338]: Will these always be 500 and 4?

**Review by rcanfield2 — APPROVED** [review]

File: aopy/data/bmi3d.py (hunk 1 above)
Comment (leoscholl) [reply aopy/data/bmi3d.py]: removed

File: aopy/data/bmi3d.py (hunk 2 above)
Comment (leoscholl) [reply aopy/data/bmi3d.py:337]: no, those are the defaults, but the line below should update them with user input

**Pull request merged**